src/main/java/edu/ucsb/cs156/team02/controllers/UCSBSubjectController.java

```diff
@@ -0,0 +1,92 @@
+package edu.ucsb.cs156.team02.controllers;
+
+import edu.ucsb.cs156.team02.entities.UCSBSubject;
+import edu.ucsb.cs156.team02.models.CurrentUser;
+import edu.ucsb.cs156.team02.repositories.UCSBSubjectRepository;
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiParam;
+import lombok.extern.slf4j.Slf4j;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+import javax.validation.Valid;
+import java.util.Optional;
+
+@Api(description = "API to handle CRUD operations for UCSB Subjects database")
+@RequestMapping("/api/UCSBSubjects")
+@RestController
+@Slf4j
+public class UCSBSubjectController extends ApiController {
+    @Autowired
+    UCSBSubjectRepository subjectRepository;
+
+    @Autowired
+    ObjectMapper mapper;
+
+    @ApiOperation(value = "List all UCSB Subjects")
+    @GetMapping("/all")
```

Comment:
If you want both Users and Admins to be able to use this API add 
`@PreAuthorize("hasRole('ROLE_USER') || hasRole('ROLE_ADMIN')")`

(same for all other endpoints)

Reply:
Fixed